### PR TITLE
Update for Play 2.5

### PR DESCRIPTION
**Pull request by rmmeans**
- Updated dependencies for Play 2.5.x
- Updated tests due to deprecated changes in Play 2.5.x
- Updated readme for 2.5 as well as updated the version available for 2.4 in install guide

### CODE REVIEW

File: redis/build.sbt [hunk 1]
```diff
@@ -1,18 +1,17 @@
...
-scalaVersion := "2.11.6"
-crossScalaVersions := Seq("2.11.6", "2.10.5")
+scalaVersion := "2.11.8"
 javacOptions ++= Seq("-source", "1.8", "-target", "1.8", "-Xlint:unchecked", "-encoding", "UTF-8")
 scalacOptions += "-deprecation"
 
 libraryDependencies ++= Seq(
-  "com.typesafe.play"         %% "play"               % "2.4.0"     % "provided",
-  "com.typesafe.play"         %% "play-cache"         % "2.4.0",
+  "com.typesafe.play"         %% "play"               % "2.5.0"     % "provided",
```
Comment (gmethvin) [inline redis/build.sbt]: Maybe we should upgrade to 2.5.3 here?

**Comment by rmmeans** [comment]
I will revisit this in the AM - fighting travis with the cross build right now. I did switch it to play 2.5.3 per your suggestion.

**Comment by rmmeans** [comment]
All fixed up. Let me know if you want anything more for this PR.

**Comment by rmmeans** [comment]
Can this be released and published as 2.5.0?

**Comment by rmmeans** [comment]
@gmethvin or @jroper - Can one of you publish this update so we can pull it as `"com.typesafe.play.modules" %% "play-modules-redis" % "2.5.0"`? The readme was updated to reflect this in this PR.

**Pull request merged**